WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m

```diff
@@ -526,6 +531,16 @@ - (void)setAvatarForComment:(Comment *)comment forCell:(ReaderCommentCell *)cell
     }];
 }
 
+- (void)checkIfCanComment
+{
+    self.canComment = self.post.commentsOpen && self.isLoggedIn;
+}
+
+- (void)checkIfLoggedIn
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+    self.isLoggedIn = [[[AccountService alloc] initWithManagedObjectContext:context] defaultWordPressComAccount] != nil;
```

Comment:
Agreed! I'll open an issue to create a "isLoggedIn" helper (on WPAccount maybe?) and we can scrub all the things.